jurity/recommenders/ctr.py

```diff
@@ -14,13 +14,50 @@
 class CTR(_BaseRecommenders):
     """Click-through rate
 
+    Three supported estimation methods:
+
+    1. Matching
     Calculates the CTR using a direct matching method. That is, CTR is only calculated for instances where the
     actual item the user has seen matches the recommendation.
+
+    2. Inverse Propensity Score (IPS)
+    Calculates the IPS, an estimate of CTR with a weighted correction based on how likely an item was to be recommended
+    by the historic policy if the user saw the item in the historic data.
+
+    ..math::
+        IPS = \frac{1}{n} \sigma \frac{r_a \times I(\hat{a} = a}{p(a|x,h)}
+
+    In this equation:
+    * n is the total size of the test data
+    * r_a is the observed reward
+    * hat{a} is the recommended item
+    * I(\hat{a} = a} is a boolean of whether the user-item pair has historic data
+    * p(a|x,h) is the probability of the item being recommended for the test context given the historic data
+
+    3. Doubly Robust Estimation (DR)
+    Calculates the DR, an estimate of CTR that combines the directly predicted values with a correction based on how
+    likely an item was to be recommended by the historic policy if the user saw the item in the historic data.
+
+    ..math::
+        DR = \frac{1}{n} \sigma (\hat{r_a} + \frac{(r_a -\hat{r_a}) I(\hat{a} = a}{p(a|x,h)})
+
+    In this equation:
+    * n is the total size of the test data
+    * r_a is the observed reward
+    * \hat{r_a} is the predicted reward
+    * hat{a} is the recommended item
+    * I(\hat{a} = a} is a boolean of whether the user-item pair has historic data
+    * p(a|x,h) is the probability of the item being recommended for the test context given the historic data
+
+    At a high level, doubly robust estimation combines a direct estimate with an IPS-like correction if historic data is
+    available. If historic data is not available, the second term is 0 and only the predicted reward is used for the
+    user-item pair.
     """
 
     def __init__(self, click_column: str, k: Optional[int] = None, user_id_column: str = Constants.user_id,
                  item_id_column: str = Constants.item_id, value_column: Optional[str] = None,
-                 n_items: Union[int, str] = None, n_sampled: Union[int, str] = None):
+                 estimation: Optional[str] = 'matching', propensity_column: Optional[str] = Constants.propensity,
```

Comment:
Probably don't want to make `estimation` optional.

Reply:
I was thinking optional as in there's a default so it's optional for the user to specify, but you're right, that's not what optional means 😆